apps/webservice/src/app/workspaces/create/page.tsx

```diff
@@ -23,42 +25,42 @@ import {
   FormItem,
   FormLabel,
   FormMessage,
-  useForm,
+  FormRootError,
 } from "@ctrlplane/ui/form";
 import { Input } from "@ctrlplane/ui/input";
 
 import { api } from "~/trpc/react";
 
-const workspaceForm = z.object({
-  name: z.string().min(3).max(30),
-  slug: z
-    .string()
-    .min(3)
-    .max(30)
-    .refine((slug) => slug === slugify(slug, { lower: true }), {
-      message: "Must be a valid URL",
-    }),
-});
+const workspaceForm = z.object(workspaceSchema.shape);
+type WorkspaceFormValues = z.infer<typeof workspaceForm>;
 
 export default function WorkspaceJoin() {
   const { data: session } = useSession();
   const create = api.workspace.create.useMutation();
-  const form = useForm({
-    disabled: create.isPending,
-    schema: workspaceForm,
+  const router = useRouter();
+
+  const form = useForm<WorkspaceFormValues>({
+    resolver: zodResolver(workspaceForm),
     defaultValues: { name: "", slug: "" },
   });
 
-  const { name } = form.watch();
-  useEffect(
-    () => form.setValue("slug", slugify(name, { lower: true })),
-    [form, name],
-  );
+  const { handleSubmit, watch, setValue, setError } = form;
 
-  const router = useRouter();
-  const onSubmit = form.handleSubmit(async (data) => {
-    await create.mutateAsync(data);
-    router.push(`/${data.slug}`);
+  watch((data, { name: fieldName }) => {
```

Comment:
do you need to call this inside a useEffect?

Reply:
I will look into this.

Reply:
I don't think so, but I don't know 100% why. 
I read the [doc](https://react-hook-form.com/docs/useform/watch) and it looks like it should be used with useffect if you want to watch specific fields, but we are using it in a "field array" populated by the form type. 

They also show useFieldArray to add and remove items from the form, but I'm not doing that in this case. Our form is relatively more simple. 

<img width="746" alt="Screen Shot 2024-09-17 at 2 54 08 AM" src="https://github.com/user-attachments/assets/f3c4235d-11b4-4773-b16e-f07665221187">
<img width="710" alt="Screen Shot 2024-09-17 at 2 54 01 AM" src="https://github.com/user-attachments/assets/2c4685e9-fad2-42b6-b76e-a56636dfd495">